File: 206. Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* dummy = new ListNode(-1);
        ListNode* nextnode = NULL;
        while (head)
        {
            ListNode* temp = head;
            head = head->next;
            temp->next = nextnode;
            dummy->next = temp;
            nextnode = temp;
        }
        return dummy->next;
    }
};

File: 206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy = ListNode(-1)
        nextnode = None
        while head:
            temp = head
            head = head.next
            temp.next = nextnode
            dummy.next = temp
            nextnode = temp
        return dummy.next
